service/storage_proxy.cc

```diff
@@ -2179,8 +2178,7 @@ future<> storage_proxy::mutate_counters(Range&& mutations, db::consistency_level
 storage_proxy::paxos_participants
 storage_proxy::get_paxos_participants(const sstring& ks_name, const dht::token &token, db::consistency_level cl_for_paxos) {
     keyspace& ks = _db.local().find_keyspace(ks_name);
-    locator::abstract_replication_strategy& rs = ks.get_replication_strategy();
-    std::vector<gms::inet_address> natural_endpoints = rs.get_natural_endpoints(token);
+    std::vector<gms::inet_address> natural_endpoints = get_natural_endpoints_without_node_being_replaced(ks, token);
```

Comment:
Don't we also need to change storage_proxy::get_live_endpoints() so that reads don't try to read from the pending (replacing) node as if it was the replaced node?

Reply:
I do not know for sure. Let's ask people who is familiar with service/storage_proxy.cc @gleb-cloudius ?

Reply:
A read should only contact those replicas that at least potentially have up-to-date data. Does a node been replaced is reported as dead or alive? If the later it should not be contacted during read.

Reply:
The node being replaced can be dead (when the replacing node is started with a different ip) or alive (when the replacing node is started  with the same IP address as the node being replaced). I think we should use get_natural_endpoints_without_node_being_replaced in get_live_endpoints to filter out the node being replaced.